File: ios/baseball/baseball/Domain/Entity/GameInfo.swift
//
//  GameInfo.swift
//  baseball
//
//  Created by 이다훈 on 2021/05/07.
//

import Foundation

struct GameInfo: Codable {
    private(set) var scores: Score
    private(set) var innings: Innings
    private(set) var awayPlayers: Players
    private(set) var homePlayers: Players
    
    init() {
        self.scores = Score(awayScore: 0, homeScore: 0)
        self.innings = Innings(home: [], away: [])
        self.awayPlayers = Players()
        self.homePlayers = Players()
    }
}

File: be/server/src/main/java/com/baseball/domain/GameInfo.java
package com.baseball.domain;

import com.baseball.domain.player.Players;

public class GameInfo {
    private final Score scores;
    private final Innings innings;
    private final Players awayPlayers;
    private final Players homePlayers;

    public GameInfo(Score scores, Innings innings, Players awayPlayers, Players homePlayers) {
        this.scores = scores;
        this.innings = innings;
        this.awayPlayers = awayPlayers;
        this.homePlayers = homePlayers;
    }

    public Score getScores() {
        return scores;
    }

    public Innings getInnings() {
        return innings;
    }

    public Players getAwayPlayers() {
        return awayPlayers;
    }

    public Players getHomePlayers() {
        return homePlayers;
    }
}
